### ISSUE TITLE
SharePoint not authenticating "No valid OAuth 2.0 authentication session exists" 

### ISSUE THREAD

**Issue by kamsalisbury** [cisagov/ScubaGear#1311]
# 🐛 Summary #

I can "Import-Module -Name Microsoft.Online.SharePoint.PowerShell -UseWindowsPowerShell"
and
"Connect-SPOService -Url https://myorg-admin.sharepoint.com/"
manually.
SCuBA 1.30 produces "Error establishing a connection with sharepoint. No valid OAuth 2.0 authentication session exists." 
Even if I manually authenticate to SharePoint Online prior to "Invoke-Scuba -M365Environment gcc -OPAPath "path-to-opa.exe"

## To reproduce ##

1. Follow instructions to "Install-Module -Name ScubaGear"
2. Initialize-SCuBA -NoOPA
3. Download OPA exe
4. "Invoke-Scuba -M365Environment gcc -OPAPath "path-to-opa.exe"

## Expected behavior ##

I expected the prompt to authenticate to SharePoint online but that did not occur - or is not easy to discern that one of the multiple authentication prompts were for SharePoint online. Then the error occurs, indicating that SharePoint authentication did not occur. 

I suspect "Connect-IPPSSession -UserPrincipalName login@fqdn.tld" is also missing and explains why the PowerPlatform check is not running either.

**Comment by kamsalisbury** [comment]
Thank you for your comment and trying to help.
I apologize for not adding the context: My environment is modern authentication with MFA required.
PowerShell 7.4.3
I turned on "Debug-SCuBA"
I re-executed "Initialize-SCuBA" and it reported all requirements met with latest versions.
"Invoke-Scuba -M365Environment gcc -OPAPath "path-to-opa.exe -ProductNames *"
produces errors:
... The term 'Add-PowerAppsAccount' is not recognized as a name of a cmdlet...
... The term 'Connect-SPOService' is not recognized as a name of a cmdlet...
So, I exited that 7.4.3 session.
Started a fresh 5.1.20348.2652 PowerShell session.
"Install-Module -Name ScubaGear -Scope CurrentUser"
"Initialize-SCuBA"
"Invoke-Scuba -M365Environment gcc -OPAPath "path-to-opa.exe -ProductNames *"
I am happy to report that this SCuBA run included SharePoint and PowerPlatform this time. No authentication issues.
I think my issue of being able to successfully execute SCuBA is resolved by sticking to PowerShell 5.1.x
For github issues, my environment points out issues with the PowerShell 7.4.x scenario.
